pkg/util/libp2p/libp2p.go

```diff
@@ -31,6 +32,24 @@ func NewDummyHost(numericID int, sourceAddr t.NodeAddress) host.Host {
 	return h
 }
 
+func NewFreeHostAddr() (m multiaddr.Multiaddr) {
+	var a *net.TCPAddr
+	var err error
+	a, err = net.ResolveTCPAddr("tcp", "localhost:0")
+	if err == nil {
+		var l *net.TCPListener
+		if l, err = net.ListenTCP("tcp", a); err == nil {
+			defer l.Close() // nolint
+			addr := l.Addr().(*net.TCPAddr).Port
```

Comment:
Cosmetic: Calling a port `addr` is a bit deceiving.

Reply:
fixed